File: Classroom/04.03/nastyaArray.cpp
// https://codeforces.com/problemset/problem/992/A

#include <bits/stdc++.h>

using namespace std;

int main(){
    
    int size;
    cin >> size;
    set<int> arr;

    for(int i = 0 ; i < size ; i++){
        int aux; cin >> aux;
        arr.insert(aux);
    }

    arr.find(0) != arr.end() ? cout << arr.size() - 1 << "\n" : cout << arr.size() << "\n";

    return 0;
}

File: Classroom/04.03/nastyaArray.py
# https://codeforces.com/problemset/problem/992/A

zeros = False
size = input()
myset = set()

arr = map(int, input().split())

for i in arr:
    myset.add(i)

print(len(myset) - 1) if 0 in myset else print(len(myset))
